src/Opw.PineBlog.EntityFrameworkCore/BlogEntityDbContext.cs

```diff
@@ -23,5 +31,162 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 entityType.SetTableName($"PineBlog_{entityType.GetTableName()}");
             }
         }
+
+        public Task<Result<int>> AddPostAsync(Post post, CancellationToken cancellationToken)
+        {
+            Posts.Add(post);
+            return SaveChangesAsync(cancellationToken);
```

Comment:
`Add` and` SaveChangesAsync` should be split between the repository and the unit of work

Reply:
The `SaveChangesAsync` seems to be an implementation detail specifically for EF Core. I'm unsure how the regular Mongo implementation would look, though they do appear to have transactions.

Reply:
MongoDb does not have this concept, changes are saved immediately when adding or updating, but to have to option of working with transaction in the future we can just have a dummy implementation for `SaveChanges`